### ISSUE TITLE
Add a11y id to "Sync and Save Data" button in synced tabs view

### ISSUE THREAD

**Issue by data-sync-user** [mozilla-mobile/firefox-ios#15556]
The "Sync and Save Data" button on the synced tabs view currently doesn’t have an accessibility id. This causes the UITest `NavigationTest` `testTapSignInShowsFxAFromRemoteTabPanel` to fail at the following line:

`waitForExistence(app.buttons["EmailSignIn.button"], timeout: TIMEOUT)`

┆Issue is synchronized with this [Jira Task](https://mozilla-hub.atlassian.net/browse/FXIOS-7007)
